### PR TITLE
Clipboard: add new-style metadata that survives browser sanitize

### PR DESCRIPTION
**Pull request by vmiklos**
- this is a first step towards fixing https://github.com/CollaboraOnline/online/issues/8023
- this doesn't switch to the new API yet, but adds a new markup to recognize if the clipboard has our own HTML or something else
- the new markup is better, because <meta> tags are removed by the browser when using the new navigator.clipboard API, and this one is not
- this is meant to be a safe set of first commits, we already write the new markup, but we don't remove the old one yet, e.g. Android needs updating still
- on the other hand, this already provides enough markup so src/map/Clipboard.js can start experimenting with the new API, including the internal paste codepath

### CODE REVIEW

**Comment by vmiklos** [comment]
> https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-23.05/9640/consoleFull

but that suite has one known-unstable test and here the insert hyperlink one failed. Fixed now so that the content of <style> doesn't leak into the plain text selection result.

**Comment by vmiklos** [comment]
@caolanm could you please review this? Thanks.

This is meant to be a safe subset, we just change the marker format, no actual use of the new clipboard stuff yet. https://github.com/CollaboraOnline/online/issues/8023#issuecomment-1898616860 shows how the old vs new markup looks like.

File: browser/src/map/Clipboard.js [hunk 1]
```diff
@@ -115,15 +119,15 @@ L.Clipboard = L.Class.extend({
 	_originWrapBody: function(body, isStub) {
 		var encodedOrigin = encodeURIComponent(this.getMetaURL());
 		var text =  '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">\n' +
-		            '<html>\n' +
+		            '<div id="meta-origin" data-coolorigin="' + encodedOrigin + '"><html>\n' +
```
Comment (caolanm) [inline browser/src/map/Clipboard.js:122]: It isn't actually valid html though, right, so could stop working if the sanitizing gets tightened up?

reminds me of the bodge for similar with gtk3 by adding an extra mimetype to identify that the clipboard content was coming from "ourself": https://cgit.freedesktop.org/libreoffice/core/commit/?id=88f7aae022bedd61588424a11bbc033217ba4e43

**Review by caolanm — APPROVED** [review]

File: browser/src/map/Clipboard.js (hunk 1 above)
Comment (vmiklos) [reply browser/src/map/Clipboard.js:122]: Oh wow, I thought desktop is nice & simple, given we have a reliable way of getting notified of the ownership change & only browser is poor. Not. :-)

In any case, good point with the invalid HTML, I'll look at that. We had the markup in <head>, that is removed, so the first thought was to do this toplevel, which gives valid output currently, and I didn't tune this further. I'll check if it's possible to do this inside <body>, that would be very similar, but would meet both the sanitize and the validity requirements, I think.

**Pull request merged**